### DISCUSSION TITLE
Nessus Scan issue w/ Security Onion 2

### DISCUSSION THREAD

**Discussion by Mr2Ez** [Security-Onion-Solutions/securityonion#6628] [Unsupported Versions] [answered]
Hope all is well.

Having issues getting a credentialed vulnerability scan using Nessus professional against a Security onion 2 node (most recent version). Specifically, I'm able to SSH to the node but I feel as though one of the capabilities (maybe OSSEC?) is preventing me from getting the results I need. I checked the Alerts tab and noticed a few things that were triggered by my scan, however, I don't know where to check to see what's the cause. Does anyone have suggestions? Thanks in advance. I can try to upload screenshots if it would help.

v/r,

Jordan

**Comment by darth-drew** [comment, marked as answer]
Be sure to whitelist your scanner address as indicated below:

https://docs.securityonion.net/en/2.3/wazuh.html?highlight=wazuh#active-response

Keep in mind this will only work on the manager, so you may want to copy ossec.conf to your other nodes and restart wazuh.

Also keep in mind that ssh comes prehardened: https://docs.securityonion.net/en/2.3/ssh.html?highlight=ssh-harden#hardening  You may need to update either the scanner configs or your sshd configs to compensate.

**Reply by Mr2Ez** [reply]
Thank you for responding! Trying this now and I'll let you know how it goes!

**Reply by Mr2Ez** [reply]
![ss_wazuh_whitelist](https://user-images.githubusercontent.com/20463554/146629176-d3f166bb-2b16-4086-9268-56a17dd3331f.PNG)
![ss_kex](https://user-images.githubusercontent.com/20463554/146629179-cf09cee8-1ff9-4f4f-b36f-b0d8f647d00b.PNG)

Unfortunately, your suggestions didn't work (unless I implemented something wrong). I added screenshots to show my kex algorithms on the manager side, as well as my wazuh whitelist. Did I do something wrong? Receiving the following error in Nessus which leads me to believe its kex but I changed the values already. 

"Unsupported key exchange algorithm : diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256"

So confused/frustrated.

**Reply by dougburks** [reply]
You can take a look at `/usr/sbin/so-ssh-harden` to see the changes that it makes to `/etc/ssh/sshd_config`. You could try comparing your sshd_config to a default sshd_config. You could also try removing the lines beginning with `ciphers`, `kexalgorithms`, `macs`, and `hostkeyalgorithms` which should set those options back to defaults.

**Reply by Mr2Ez** [reply]
Sorry for the delayed response, went out of town. Thank you both very much for responding and taking the time to help me with my issue. I ended up copying a base ssh conf and it solved my issue. I also disabled wazuh but I'm not sure it was necessary. Thanks again!

**Comment by darth-drew** [comment]
Have you tried restarting/disabling wazuh and commenting out those restrictions in your sshd config? It may also be worth tailing your var/log/secure (or equivalent) on both sides and looking for clues there

**Reply by Mr2Ez** [reply]
I've tried disabling Wazuh (thought this was it for sure), but same result. Going to look again at the sshd conf and tail logs. thanks.
